File: pset6/readability/readability.py
from cs50 import get_string
import re

# reading the text
t = get_string("Text: ")

# couting sentences
t1 = re.findall('[.?!]', t)
sentences = len(t1)

# counting words
t1 = re.findall("[^\W\d_]+(?:['][^\W\d_]+)*", t)  # finding all alphabetic words
words = len(t1)

# counting letters
t1 = re.findall('\w', str(t1))  # finding all alphabetic letters
letters = len(t1)

# apply teh Coleman-Liau index
l = letters/words*100
s = sentences/words*100
grade = 0.0588 * l - 0.296 * s - 15.8

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade ", round(grade))

File: Problem-Set-2/sTaissa-cs50-problems-2020-x-readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //reading the text and declaring variables
    string t = get_string("Texto: ");
    int n = strlen(t);
    float letters = 0, words = 0, sentences = 0;

    //counting letters, words and sentences
    for (int i = 0; i < n; i++)
    {
        if (isalpha(t[i]))
        {
            letters++;
        }
        else if (isspace(t[i]) || i == n - 1)
        {
            words++;
        }
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            sentences++;
        }
    }

    //apply the Coleman-Liau index
    float grade = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8;

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(grade));
    }
}